### PR TITLE
Feature/platfowner/new feature

### PR DESCRIPTION
**Pull request by platfowner**
Summary:
- Increase efficiency of state version finalization / deletion operations by using numRef values of state nodes.
- Log deleted state nodes when state version is deleted.
- Simplify some version-finalizing code.

#143

### CODE REVIEW

File: db/state-node.js [hunk 1]
```diff
@@ -98,12 +123,30 @@ class StateNode {
...
+  }
+
+  increaseNumRef() {
+    this.numRef++;
+  }
+
+  decreaseNumRef() {
+    if (this.numRef > 0) {
+      this.numRef--;
+    }
```
Comment (liayoo) [inline db/state-node.js:145]: maybe log/throw error if this.numRef <= 0?

File: db/state-manager.js [hunk 2]
```diff
@@ -134,23 +133,25 @@ class StateManager {
...
     }
+    let numDeletedNodes = null;
     if (FeatureFlags.enableStateVersionOpt) {
-      deleteStateTreeVersion(root, version);
+      numDeletedNodes = deleteStateTreeVersion(root, version);
     } else {
-      deleteStateTree(root);
+      numDeletedNodes = deleteStateTree(root);
     }
+    logger.info(`[${LOG_HEADER}] Deleted ${numDeletedNodes} sate nodes.`);
```
Comment (liayoo) [inline db/state-manager.js:154]: typo? (state)

**Review by liayoo — APPROVED** [review]
LGTM with some comments. Thanks!

File: db/state-manager.js (hunk 2 above)
Comment (platfowner) [reply db/state-manager.js]: fixed. thanks.

**Review by minsulee2 — APPROVED** [review]
LGTM.
Is it good time to version up?!

File: db/state-node.js (hunk 1 above)
Comment (platfowner) [reply db/state-node.js:151]: Done. Thanks!

**Comment by platfowner** [comment]
> LGTM.
> Is it good time to version up?!

Yes, let me prepare a PR for version up.

**Pull request merged**